lib/vrl/compiler/src/type_def.rs

```diff
@@ -19,6 +22,25 @@ pub struct TypeDef {
     kind: KindInfo,
 }
 
+impl Sub<Kind> for TypeDef {
+    type Output = Self;
+
+    /// Removes the given kinds from this type definition.
+    fn sub(mut self, other: Kind) -> Self::Output {
+        let kind = match self.kind {
+            KindInfo::Unknown => KindInfo::Unknown,
+            KindInfo::Known(kinds) => {
+                // Note, it would be much nicer to use retain here, but that is unfortunately still in nightly.
```

Comment:
```suggestion
```

While true, I don't think there's much value in adding comments like these into the source. If/when `retain` stabilizes, I'm sure there'll be a Clippy warning to help us solve these issues, and if not, we can fix them when we encounter them.